packages/wrangler/src/config/validation.ts

```diff
@@ -3200,6 +3200,47 @@ const validateConsumer: ValidatorFn = (diagnostics, field, value, _config) => {
 	return isValid;
 };
 
+const isValidDate = (value: string): boolean => {
+	const data = new Date(value);
+	return !isNaN(data.getTime());
+};
+
+const validateCompatibilityDate: ValidatorFn = (diagnostics, field, value) => {
+	if (value === undefined) {
+		return true;
+	}
+
+	if (typeof value !== "string") {
+		diagnostics.errors.push(
+			`Expected "${field}" to be of type string but got ${JSON.stringify(value)}.`
+		);
+		return false;
+	}
+
+	let isValid = true;
+
+	if (
+		value.includes("–") || // en-dash
+		value.includes("—") // em-dash
+	) {
+		isValid = false;
+
+		diagnostics.errors.push(
+			`"${field}" field should use hyphens (-) rather than en-dashes (—) or em-dashes (–).`
```

Comment:
Is there a better way to demonstrate this, given that all three look identical in monospace fonts? Maybe there isn't, but I thought I would challenge it just because it reads very strangely in the terminal.

Reply:
not that I can tell I'm afraid. I totes get what you mean, and I did look into it, but it doesn't seem to be possible for us to control how the user's terminal will display these things.

Reply:
NP! Thanks for looking into it.